purchase_order_owner/models/purchase_order.py

```diff
@@ -3,16 +3,13 @@
 
 from odoo import fields, models
 
-from odoo.addons.purchase.models.purchase import PurchaseOrder
-
 
 class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
 
     owner_id = fields.Many2one(
         "res.partner",
         "Assign Owner",
-        states=PurchaseOrder.READONLY_STATES,
```

Comment:
A corresponding adjustment should be made in the view (`purchase_order_form`), adding `readonly="state in ['cancel', 'done', 'purchase']"`.